Source/Syrup/Player/SyrupPlayerCharacter.h

```diff
@@ -34,8 +37,28 @@ class SYRUP_API ASyrupPlayerCharacter : public ACharacter
 	
 	/* /\ Components /\ *\
 	\* ---------------- */
+	
+	/* ----------------- *\
+	\* \/  Resources  \/ */
+private:
+	//Energy of the Character
+	UPROPERTY(EditAnywhere, Category = "Resources")
+	int Energy;
+
+	/* /\  Resources  /\ *\
+	\* ----------------- */
+
+	/* ------------------------ *\
+	\* \/  Plant Management  \/ */
+public:
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = "Plant")
+	int PlantSelected;
```

Comment:
Should be private and have a getter and setter function to ensure the variable is always within the range of the valid indices of PlantTypes.

Also needs comment.